curator/src/main/java/edu/illinois/cs/cogcomp/curator/CuratorAnnotatorService.java

```diff
@@ -161,6 +185,18 @@ public TextAnnotation createAnnotatedTextAnnotation(String corpusId, String text
     public TextAnnotation createAnnotatedTextAnnotation(String corpusId, String textId,
             String text, Set<String> viewNames) throws AnnotatorException {
         TextAnnotation ta = createBasicTextAnnotation(corpusId, textId, text);
+
+        if (annotationCache != null && annotationCache.contains(ta)) {
+            ta = annotationCache.getTextAnnotation(ta);
+            // clone the views
+            HashSet<String> views = new HashSet<>(viewProviders.keySet());
+            views.removeAll(ta.getAvailableViews());
+            if(views.size() == 0) {
+                // it contains the required views; otherwise continue
+                return ta;
+            }
+        }
+
```

Comment:
Yeah, currently in master, the only place we have caching is inside `addView`. 
Suppose you want to add a bunch of views (like pos, ner, parse, srl). Even if you have it in the cache, you'd have to call this functions for each of these required views. By checking the cache in an earlier stage, I prevent this behavior: if the item is in the cache, we never call `addView` functions and get the result directly. 

If our caching was in the level of Views, I'd totally be for keeping it as it is (i.e. caching in `addView`). But currently caching is done in the level of `TextAnnotation`.